controllers/cloud.redhat.com/clowdenvironment_reconciliation.go

```diff
@@ -0,0 +1,489 @@
+//If it is then we enter into the SKIPRECONCILE corner case
+//which is a special case where the reconciliation controller bails early
+//from reconcile WIHTOUT returning an error
+func (r *ClowdEnvironmentReconciliation) isTargetNamespaceMarkedForDeletion() (ctrl.Result, error) {
+	ens := &core.Namespace{}
+	if getNSErr := r.client.Get(*r.ctx, types.NamespacedName{Name: r.env.Status.TargetNamespace}, ens); getNSErr != nil {
+		return ctrl.Result{Requeue: true}, getNSErr
+	}
+
+	if ens.ObjectMeta.DeletionTimestamp != nil {
+		r.log.Info("Env target namespace is to be deleted - skipping reconcile")
+		return ctrl.Result{}, errors.New(SKIPRECONCILE)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *ClowdEnvironmentReconciliation) runProviders() (ctrl.Result, error) {
+	provider := r._makeProvider()
+	provErr := runProvidersForEnv(*r.log, provider)
+
+	if provErr != nil {
+		r.log.Info("Prov err", "err", provErr)
+		if setClowdStatusErr := SetClowdEnvConditions(*r.ctx, r.client, r.env, crd.ReconciliationFailed, provErr); setClowdStatusErr != nil {
+			r.log.Info("Set status error", "err", setClowdStatusErr)
+			return ctrl.Result{Requeue: true}, setClowdStatusErr
+		}
+		return ctrl.Result{Requeue: true}, provErr
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *ClowdEnvironmentReconciliation) _makeProvider() providers.Provider {
+	return providers.Provider{
+		Ctx:    *r.ctx,
+		Client: r.client,
+		Env:    r.env,
+		Cache:  r.cache,
+		Log:    *r.log,
+	}
+}
+
+func (r *ClowdEnvironmentReconciliation) applyCache() (ctrl.Result, error) {
+	cacheErr := r.cache.ApplyAll()
+	if cacheErr != nil {
+		r.log.Info("Cache error", "err", cacheErr)
+		if setClowdStatusErr := SetClowdEnvConditions(*r.ctx, r.client, r.env, crd.ReconciliationFailed, cacheErr); setClowdStatusErr != nil {
+			r.log.Info("Set status error", "err", setClowdStatusErr)
+			return ctrl.Result{Requeue: true}, setClowdStatusErr
+		}
+		return ctrl.Result{Requeue: true}, cacheErr
+	}
+	return ctrl.Result{}, nil
+}
+
+//Sets info for the apps in the environment
+//This method is the step and contains most of the error handling, logging etc
+//The full implementaiton is pushed out into another method
+func (r *ClowdEnvironmentReconciliation) setAppInfo() (ctrl.Result, error) {
+	if setAppErr := r._setAppInfoImplementation(); setAppErr != nil {
+		r.log.Info("setAppInfo error", "err", setAppErr)
+		if setClowdStatusErr := SetClowdEnvConditions(*r.ctx, r.client, r.env, crd.ReconciliationFailed, setAppErr); setClowdStatusErr != nil {
+			r.log.Info("Set status error", "err", setClowdStatusErr)
+			return ctrl.Result{Requeue: true}, setClowdStatusErr
+		}
+		return ctrl.Result{Requeue: true}, setAppErr
+	}
+	return ctrl.Result{}, nil
+}
+
+//Performs the actual business logic for the setAppInfo step
+//TODO this is a bear of a method. Factor this out into pieces.
+func (r *ClowdEnvironmentReconciliation) _setAppInfoImplementation() error {
+	p := r._makeProvider()
```

Comment:
You don't actually need a provider object now, you already have everything you need on `r` :)

Reply:
I would request that we do that later because this method is a straight copy and paste from the original code and I don't understand what it is doing. Considering the size and complexity of the method I'd prefer to leave it as is (which is exactly what was in clowder before) and then we tackle it later.

Reply:
Refactored. You were right, way easier than I expected. Just deleted the provider and let the red squiggles tell me what to fix.